internal/common/function/function.go

```diff
@@ -139,6 +140,51 @@ func GenerateVeleroBackupName(namespace, nabName string) string {
 	return veleroBackupName
 }
 
+// GenerateVeleroBackupNameWithUUID generates a Velero backup name based on the provided namespace and NonAdminBackup name.
+// It includes a UUID suffix. If the name exceeds the maximum length, it truncates nabName first, then namespace.
+func GenerateVeleroBackupNameWithUUID(namespace, nabName string) string {
```

Comment:
I assume whatever calls this func is going to call it again if on create it found a duplicate velero backup name?

Reply:
No. This function is going to be called once and the generated UUID will be unique. This UUID will get stored in both NAB object and other objects associated with that NAB such as Backup.

Reply:
How can you be sure?
Didn't we say we wanna retry on conflict?

Either way that should be done on create call failure due to already existing backup of the same name.

Reply:
> [simply use our generated UUID as the Backup object name with retry generate](https://github.com/migtools/oadp-non-admin/issues/90#issuecomment-2376188425)

> [Yeah, a retry on velero backup name seems a good approach](https://github.com/migtools/oadp-non-admin/issues/90#issuecomment-2376831858)

Reply:
@kaovilai that is exactly how this should be used:

1. Whoever calls this function implements retry logic
2. The NonAdminBackup calls this function to get the Backup name as UUID - to make it easier let's call it UUID_1 if it fails due to unlikely event of collision it does retry.
3. Once 2 succeed the Backup is generated with the name that corresponds to this name. Moreover the label is added with exactly same name (UUID_1).
4. The UUID_1 is also stored in NonAdminBackup object as label value.
5. When next reconcile comes to NonAdminBackup it checks if there is label with some UUID. We get in our case UUID_1 so we can get the Velero Backup object with UUID_1. It works vice-versa, whenever Backup object requires to update NAB object it can be identified by this UUID_1.

Reply:
Sounds good.